a3update.py

```diff
@@ -122,25 +141,9 @@ def lowercase_workshop_dir(path: str):
 
 
 def check_workshop_mod(mod_id: str):
-    response = request.urlopen("{}/{}".format(WORKSHOP_CHANGELOG_URL, mod_id)).read().decode("utf-8")
-    mod_name = MOD_NAME_REGEX.search(response).group(1)
-    mod_last_updated = LAST_UPDATED_REGEX.search(response)
-    path = "{}/{}".format(A3_STEAM_WORKSHOP_DIR, mod_id)
-    WORKSHOP_MODS[mod_name] = mod_id
-
-    if os.path.isdir(path) and mod_last_updated:
```

Comment:
You've removed the directory datetime check and made it so every mod is redownloaded every startup, that's no bueno. This actually may be the more valid place to download the mod with the new DOWNLOAD_ATTEMPTS so each mod gets the same download attempt individually and we can copy the mod keys to the right folder after it's downloaded. 

The current implementation bundles all the workshop mod downloads. I believe that's because steam would have to keep signing in and would cause timeouts for large number of mods trying to download too fast, but that could be fixed with the retries too? maybe...

Reply:
The reason why it was removed I believe was because after the folder is created and the download times out, it would skip the partially downloaded mod since it was "up to date". Steamcmd is an unreliable pile of crap, to be extra sure I simply removed the check, forcing a validation on every mod every time. 

It is impossible (at least with larger modsets) to do individual attempts on the mods because as you said, there's a very strict login timeout on Steam's end. The only way it can be done is to do one or two steamcmd calls and hope to do everything in a single batch.

I would like to know though, why do you consider it a big no no?